neqo-http3/src/headers_checks.rs

```diff
@@ -119,24 +126,32 @@ pub fn headers_valid(headers: &[Header], message_type: MessageType) -> Res<()> {
     }
     // Clear the regular header bit, since we only check pseudo headers below.
     pseudo_state.remove(PseudoHeaderState::Regular);
-    let pseudo_header_mask = match message_type {
-        MessageType::Response => enum_set!(PseudoHeaderState::Status),
+    let (pseudo_header_required, pseudo_header_mask) = match message_type {
+        // Responses contain only :status
+        MessageType::Response => (enum_set!(PseudoHeaderState::Status), PSEUDO_HEADER_ALL),
         MessageType::Request => {
             if method_value == Some("CONNECT") {
```

Comment:
Feel free to ignore.

```suggestion
        MessageType::Request if method_value == Some("CONNECT") => {
```

Given that the `if` `else` block spans the entire `match` arm, the above would safe a level of indentation.

I can also propose this as a follow-up.

Reply:
`rustfmt` seems to want to add the braces back around the `if`.

Reply:
So you can't do:

```rust
match whatever {
  MessageType::Request if method_value == Some("CONNECT") => { code }
  MessageType::Request => { else_clause }
}
```
?